Port statement import

steveferry opened the pull request:
Porting g11, v11, CSV (UBS, Raffaisen) parsers

nbessi committed:
Merge pull request #36 from CompassionCH/7.0-statement-import

Add swiss bank statement imports

nbessi committed:
Rebase missing statement import fixes

nbessi committed:
Remove deprecated file

nbessi committed:
Port of postfinance XML file into bank statements

nbessi committed:
Add support of scan attachement on statement import

This is a temporary solution that should be deprecated by
https://github.com/OCA/bank-statement-import/pull/21

nbessi committed:
Add statement view file

nbessi committed:
Do not import Total line

nbessi committed:
Add bank-statement-import dependencie

nbessi committed:
PEP8

nbessi committed:
Add raw file as attachment

steveferry committed:
Porting g11 parser, change acc_number if it's a bvr_adherent_number

steveferry committed:
Porting v11 parser
Needed modules l10n_ch_lsv_dd and l10n_ch_payement_slip_base_transaction_id to make it work (They're not ported!!!)

steveferry committed:
PEP8

steveferry committed:
PEP8

steveferry committed:
Porting Raffaisen parser

steveferry committed:
Porting parser Raffaisen and UBS

steveferry committed:
Porting parsers

steveferry committed:
Porting parsers - Fix journal_id problem

ecino commented:
You need to fix tests for module lsv_dd in :
l10n_ch_lsv_dd/test/lsv-dd-test.yml

You need to fix PEP8:
./l10n_ch_lsv_dd/model/bank.py:25:1: E302 expected 2 blank lines, found 1

Also, since module lsv_dd is taken out from **unported**, it would be nice to adapt the code to V8.

albertsh committed:
convert v7 to v8

albertsh committed:
update test yaml

albertsh committed:
convert module v7 to v8

Commit:
fix error in Generated direct debit files

Commit:
fix some error in convert v7 to v8

Commit:
fix some error in convert v7 to v8

Commit:
fix bug

Commit:
fix bug convert v7 to v8

Commit:
correct code for converting v7 to v8

Commit:
fix in convert v7 to v8 + PEP8

Commit:
fix in convert v7 to v8 + PEP8

Commit:
fix bug

ecino commented:
Can someone add the workinprogress label please ? We are not yet ready with this one.

steveferry committed:
FIX Three last lines were not imported

steveferry committed:
FIX Remove PDB

max3903 commented:
CLA are fixed.

steveferry committed:
FIX
Add uuid for unique ID in bank statement line
Remove view of related files (file .xml not removed)

steveferry committed:
Add the BVR picture for the statement line during reconciliation

steveferry committed:
Rename variables

steveferry committed:
Set the Postal account to None

ecino commented:
@guewen We are ready with this now, can you remove work in progress label ?

yvaucher commented on l10n_ch_account_statement_base_import/README.rst:
```diff
@@ -0,0 +1,70 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Module name
```
Put the module name here

yvaucher commented on l10n_ch_account_statement_base_import/README.rst:
```diff
@@ -0,0 +1,70 @@
+------------
+
+* Nicolas Bessi <nicolas.bessi@camptocamp.com>
+* Please Compassion put your email here.
```
@ecino to add

Emanuel Cino ecino@compassion.ch

yvaucher commented on l10n_ch_account_statement_base_import/__openerp__.py:
```diff
@@ -0,0 +1,40 @@
+##############################################################################
+
+{'name': "Swiss bank statements import",
+ 'version': '0.4',
```
Please put full version `8.0.0.4.0`

yvaucher commented on l10n_ch_account_statement_base_import/account_bank_statement_import.py:
```diff
@@ -0,0 +1,160 @@
+                            'datas': attachment[1],
+                        }
+                    )
+                    statement_line.related_file = [attachment.id]
```
attachment is a record set you can assign it directly

yvaucher commented on l10n_ch_account_statement_base_import/account_bank_statement_import.py:
```diff
@@ -0,0 +1,160 @@
+        for parser_class in base_parser.BaseSwissParser.__subclasses__():
+            yield parser_class(data_file)
+
+    def _create_bank_statements(self, stmts_vals):
```
Please add `@api.model`

yvaucher commented on l10n_ch_account_statement_base_import/parser/g11_file_parser.py:
```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Steve Ferry
```
Can you add a year with copyright ?

steveferry committed:
FIX

steveferry commented:
Fix made

ecino commented:
Can you rebase on OCA/8.0 ? It should remove Travis errors.

steveferry committed:
FIX travis error

ecino commented:
You need to install xlrd for the tests by adding it to travis (with pip for instance, avoid using sudo)

eLBati commented:
@steveferry please remove the ported modules from `__unported__` folder

steveferry committed:
Deleting module in __unported__

steveferry committed:
FIX travis xlrd

steveferry committed:
Fix travis error

steveferry committed:
Few fixes to match tests with class

steveferry committed:
Apply changes from OCA/bank-statement-import#15

yvaucher commented on .travis.yml:
```diff
@@ -30,7 +30,9 @@ virtualenv:
   - pip install unicodecsv
+  - pip install xlrd
   - git clone --depth=1 https://github.com/OCA/maintainer-quality-tools.git ${HOME}/maintainer-quality-tools
+  - git clone https://github.com/OCA/bank-statement-import.git ${HOME}/build/CompassionCH/bank-statement-import -b 8.0
```
To add bank-statement-import, please add it in oca_dependencies.txt

https://github.com/OCA/l10n-switzerland/blob/7.0/oca_dependencies.txt

yvaucher commented on l10n_ch_account_statement_base_import/README.rst:
```diff
@@ -0,0 +1,70 @@
+
+
+Known issues / Roadmap
+======================
```
Please remove those empty sections

yvaucher commented on l10n_ch_account_statement_base_import/README.rst:
```diff
@@ -0,0 +1,70 @@
+    :alt: License: AGPL-3
+
+Swiss bank statements import
+===========
```
Would be nice to complete this line.

steveferry committed:
Port tests for parsers

ecino commented:
Does someone know why the runbot fails on dependencies (travis is fine) ?

ecino commented on l10n_ch_account_statement_base_import/README.rst:
```diff
@@ -0,0 +1,65 @@
+
+* Nicolas Bessi <nicolas.bessi@camptocamp.com>
+* Emanuel Cino <ecino@compassion.ch>
+
```
@steveferry Add your name here

ecino commented on l10n_ch_account_statement_base_import/account_bank_statement_import.py:
```diff
@@ -0,0 +1,183 @@
+class AccountBankStatementLine(models.Model):
+    _inherit = 'account.bank.statement.line'
+
+    related_file = fields.Many2many(
```
Can a line have several attachments ? If yes, maybe we could rename this field to `related_files`.
If not, maybe we could change to `Many2one` ?

ecino commented on l10n_ch_account_statement_base_import/view/statement_view.xml:
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="utf-8"?>
```
This view is not needed anymore. It references old field (already_completed) which does not exist in v8 and adds a button (download_attachment) that has no corresponding code to be called. Please remove this file and its reference in openerp manifest.

steveferry committed:
Add view file in statement

yvaucher commented:
Superseeded by #162
